environment-dev.yml

```diff
@@ -4,9 +4,9 @@ channels:
 dependencies:
   - numpy
   - geopandas-base
-  - libgdal ==3.6.2
+  - libgdal==3.6.2
```

Comment:
It is not necessary, but on Windows the gdal version needs to be specified via an environment variable to build pyogrio. The version was pinned here so it is in sync with the line in the documentation that sets this environment variable: https://pyogrio.readthedocs.io/en/latest/install.html#conda

Reply:
I would say that a contributor can also check the version of gdal (`conda list`) to write the correct env variable (and we can mention that in the docs), instead of pinning it here. And if pinning, we should update it to 3.7